analyzers/tests/SonarAnalyzer.Test/TestCases/ExceptionsShouldBeLogged.Microsoft.Extensions.Logging.cs

```diff
@@ -0,0 +1,405 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+// Logging methods from: https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loggerextensions?view=dotnet-plat-ext-8.0
+public class TestCases
+{
+    private readonly ILogger logger = new Logger<TestCases>(new NullLoggerFactory());
+
+    public void Simple()
```

Comment:
This can be shorter and more readable if you use parameterized UTs. In my PR I've grouped the tests like this:
- one test case file for general scenarios (basic cases, method overloads, edge cases, etc.)
- [one parameterized UT method for each framework](https://github.com/SonarSource/sonar-dotnet/blob/2f28dc7c2d1e0910280327560d0afb43924e5add/analyzers/tests/SonarAnalyzer.Test/Rules/UseConstantLoggingTemplateTest.cs#L34)